block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -192,19 +224,74 @@ retrievalWorkerImpl keepAliveTimer SendActions {..} =
                     (MsgRequestBlockHeaders $ Just $ S.singleton nodeId)
                     (Conversation $ requestHeaders cont mgh nodeId)
 
+----------------------------------------------------------------------------
+-- HTTP-based block retrieval
+----------------------------------------------------------------------------
+
+-- | Try to download and apply blockchain dumps, starting from the given
+-- epoch. Halts after having downloaded enough dumps for the current slot to
+-- become known.
+downloadBlockDump
+    :: forall ctx m. BlockWorkMode ctx m
+    => EpochIndex  -- ^ The first epoch for which to download blocks
+    -> String      -- ^ URL of a folder with block dumps
+    -> m ()
+downloadBlockDump epoch dumpUrl = whenNothingM_ getCurrentSlot $ do
+    let epochUrl = dumpUrl <//> format "epoch{}.cbor.lzma" (toInteger epoch)
+    logInfo ("Downloading blockchain dump for epoch "+|epoch|+
+             " from "+|epochUrl|+"")
+    isSuccess <- modifyStateLock HighPriority "tryDownload" $ \tip -> do
+        updateRecoveryHeader Nothing =<< DB.getTipHeader
+        slot <- getEpochOrSlot <$> DB.getTipHeader
+        -- When we're already in the epoch we're going to be downloading
+        -- blocks from, it's possible that we're in slot 10000 or
+        -- something and we'll have to skip 10000 blocks before we can
+        -- show any progress. In this case let's warn about it.
+        whenRight (unEpochOrSlot slot) $ \slotId ->
+            when (slotId ^. epochIndexL == epoch) $
+                logDebug ("Going to download blocks and skip until slot "
+                          +|view siSlotL slotId|+", this may take some time")
+        let applyDump
+                = decodeBlockDumpC
+               .| (C.dropWhileC ((/= tip) . view prevBlockL)
+               >> C.transPipe lift (verifyAndApplyBlocksC True))
+        -- We use Data.Conduit.Async to keep a buffer of downloaded data and
+        -- apply blocks in parallel with downloading them. The buffer size
+        -- is 16k * 512 = 8 megabytes.
```

Comment:
Where did you get these numbers from? 😒

Reply:
16 kB = I tried to remember what was the default chunk size from `Data.ByteString.Lazy` but failed (it's actually 32 kB). I'll change it.

8 MB = approximately one epoch.

Reply:
Just add some comments about it.